pymks/fmks/correlations.py

```diff
@@ -0,0 +1,87 @@
+"""MKS Correlation Module
+
+For computing auto and cross corelations under assumption
+of periodic boundary conditions using discreete fourier
+transform.
+"""
+
+
+import numpy as np
+from toolz.curried import pipe, curry
+from .func import dafftshift, dafftn, daifftn, daconj
+
+
+def faxes(arr):
+    """Get the spatial axes to perform the Fourier transform
+
+    The first axis should not have the Fourier transform
+    performed.
+
+    Args:
+      arr: the discretized array
+    Returns:
+      an array starting at 1 to n - 1 where n is the length of the
+      shape of arr
+
+    >>> faxes(np.array([1]).reshape((1, 1, 1, 1, 1)))
+    (1, 2, 3, 4)
+    """
+    return tuple(np.arange(arr.ndim - 1) + 1)
+
+
+def corr_master(arr1, arr2):
```

Comment:
Also add some tests for chunking. Make sure when the data is chunked on the first and last axes that the returned chunked data is as you would expect.

Reply:
added test for chunking in autocorrelation and crosscorrelation functions to make sure that the output of fft has the same chunks as the input.